### PR TITLE
fix(quantic): issues in the QuanticColoredResultBadge component fixed

### PR DESCRIPTION
**Pull request by mmitiche**
## Issues fixed in this PR:

1. [SFINT-4776](https://coveord.atlassian.net/browse/SFINT-4776): Better way to display the error in the Quantic Colored Result Badge 

issue1

2. [SFINT-4774](https://coveord.atlassian.net/browse/SFINT-4774) Error when the color property is not specified:

```
VM1546:3 Uncaught (in promise) TypeError: Cannot read properties of undefined (reading '0')
    at f (eval at t (aura_prod.js:88:77), <anonymous>:3:1102)
    at g.generateSecondaryColor (eval at t (aura_prod.js:88:77), <anonymous>:3:2068)
    at g.setBadgeColors (eval at t (aura_prod.js:88:77), <anonymous>:3:1880)
    at g.renderedCallback (eval at t (aura_prod.js:88:77), <anonymous>:3:1607)
```

3. [SFINT-4773](https://coveord.atlassian.net/browse/SFINT-4773): Text not truncated when displaying a result field in the colored result badge: 
textOverflow

## Solutions:

- Added Properties validation logic to the QuanticColoredResultBadge component:

QCRBError

- Properly truncated  text when it's too long:

error2

[SFINT-4776]: https://coveord.atlassian.net/browse/SFINT-4776?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SFINT-4774]: https://coveord.atlassian.net/browse/SFINT-4774?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[SFINT-4773]: https://coveord.atlassian.net/browse/SFINT-4773?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by nathanlb** [comment]
While this is open, are the font sizes the same between the quanticColoredResultBadge and the quanticResultBadge? I thought they looked a tad different on a screenshot

File: packages/quantic/force-app/main/default/lwc/quanticColoredResultBadge/__tests__/quanticColoredResultBadge.test.js [hunk 1]
```diff
@@ -67,67 +80,134 @@ describe('c-quantic-colored-result-badge', () => {
...
+        expect(badge.style._values['--primaryColor']).toBe(exampleColor);
+        expect(badge.style._values['--secondaryColor']).toBe('black');
+      });
     });
 
-    it(`should display the colored badge with the correct label and the default colors when the given value is an invalid HEX color value`, async () => {
-      const invalidColor = '#AAAAAAB';
-      const element = createTestComponent(invalidColor, defaultLabel);
-      await flushPromises();
+    describe('when no field is given', () => {
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticColoredResultBadge/__tests__/quanticColoredResultBadge.test.js:143]:
```suggestion
    describe('when no result is given', () => {
```

File: packages/quantic/force-app/main/default/lwc/quanticColoredResultBadge/__tests__/quanticColoredResultBadge.test.js [hunk 2]
```diff
@@ -67,67 +80,134 @@ describe('c-quantic-colored-result-badge', () => {
...
-      expect(badge.style._values['--secondaryColor']).toBe(defaultSecondaryColor);
+        expect(errorMessage).not.toBeNull();
+        expect(console.error).toHaveBeenCalledWith(coloredResultBadgeError);
+      });
     });
 
-    it('should display the colored badge with the correct label and the default colors when the color is not given', async () => {
-      const element = createTestComponent('',defaultLabel);
-      await flushPromises();
+    describe('when no result is given', () => {
```
Comment (mmitiche) [inline packages/quantic/force-app/main/default/lwc/quanticColoredResultBadge/__tests__/quanticColoredResultBadge.test.js:159]:
```suggestion
    describe('when no field is given', () => {
```

**Comment by mmitiche** [comment]
> While this is open, are the font sizes the same between the quanticColoredResultBadge and the quanticResultBadge? I thought they looked a tad different on a screenshot

Indeed,  the font sizes  between the quanticColoredResultBadge and the quanticResultBadge were different.
just fixed that and now we have the following result:

FontsizeFixed

File: packages/quantic/force-app/main/default/lwc/quanticColoredResultBadge/quanticColoredResultBadge.js [hunk 3]
```diff
@@ -46,17 +47,57 @@ export default class QuanticColoredResultBadge extends LightningElement {
...
     // eslint-disable-next-line no-unused-expressions
-    this.hasTextToDisplay && this.setBadgeColors();
+    this.isValid && this.hasTextToDisplay && this.setBadgeColors();
   }
 
   /**
    * get the text content to display depending the provided props.
    * @returns {string}
    */
   get textToDisplay() {
```
Comment (nathanlb) [inline packages/quantic/force-app/main/default/lwc/quanticColoredResultBadge/quanticColoredResultBadge.js:99]: Could just remove this whole getter and declare hasTextToDisplay like this

```javascript
get hasTextToDisplay() {
  return !!(this.label || this.result?.raw?.[this.field]);
}
```

**Review by nathanlb — APPROVED** [review]
Very nice 👏

File: packages/quantic/force-app/main/default/lwc/quanticColoredResultBadge/quanticColoredResultBadge.js (hunk 3 above)
Comment (mmitiche) [reply packages/quantic/force-app/main/default/lwc/quanticColoredResultBadge/quanticColoredResultBadge.js:99]: sure 👌

**Pull request merged**